### ISSUE TITLE
Error while opening UI: 426 (Upgrade Required)

### ISSUE THREAD

**Issue by nadworny** [kubevious/kubevious#46]
### Describe the bug
When I proxy the UI through kubectl, the UI is loaded (although kind of empty) and I'm getting the following error:
```
index.js:16 GET http://localhost:3000/socket/?EIO=3&transport=polling&t=NKnDaco 426 (Upgrade Required)
```
All other pods are running without any errors.

Log from kubevious-ui:
```
127.0.0.1 - - [16/Oct/2020:11:49:12 +0000] "GET /socket/?EIO=3&transport=polling&t=NKnFX7v HTTP/1.1" 426 0 "http://localhost:3000/?tme=false&tmdt=RnJpIE9jdCAxNiAyMDIwIDEzOjQ5OjAyIEdNVCswMjAwIChDZW50cmFsIEV1cm9wZWFuIFN1bW1lciBUaW1lKQ==&tmd=MjQ=&tmdaf=VGh1IE9jdCAxNSAyMDIwIDEzOjQ5OjAyIEdNVCswMjAwIChDZW50cmFsIEV1cm9wZWFuIFN1bW1lciBUaW1lKQ==" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.75 Safari/537.36" "-"
```

### To Reproduce
1. Run `kubectl port-forward $(kubectl get pod -l k8s-app=kubevious-ui -n kubevious -o jsonpath="{.items[0].metadata.name}") 3000:80 -n kubevious`
1. Open localhost:3000

### Expected behavior
UI with all the information

### Screenshots
![image](https://user-images.githubusercontent.com/1425919/96254392-82082080-0fb5-11eb-8755-b53d74a2c20c.png)

### Environment Details:
 - AKS (v1.16.13)
 - Chrome 86.0.4240.75 (Official Build) (64-bit)
 - Version 0.6.36

**Comment by guillem-riera** [comment]
I am getting this issue too.

I deployed to minikube using the provided helm chart command line and version from the readme:

```
helm install kubevious kubevious/kubevious --version 0.6.36 -n kubevious

NAME: kubevious
LAST DEPLOYED: Fri Oct 16 13:52:37 2020
NAMESPACE: kubevious
STATUS: deployed
REVISION: 1
TEST SUITE: None
```

I see the same logs as @nadworny too.

Thanks for taking a look at it = )

**Comment by rubenhak** [comment]
@nadworny, @guillem-riera,
Can you guys attach logs for "kubevious" and "parser" pods? 

@nadworny, If the cluster is big with lots of pods, jobs, configmaps, etc it may take some time to bring up the UI. Just trying to find a root cause.

**Comment by nadworny** [comment]
@rubenhak here are the logs

parser: https://gist.github.com/nadworny/11f232ec92863022268904db51b2e05a

kubevious: https://gist.github.com/nadworny/c9a666f8e1c77d1f52255dfcb42fa5b1

**Comment by nadworny** [comment]
Ok as suggested, I waited a bit and started the UI again - it works! 👍 

I'm getting this exception but I guess it's irrelevant:
`index.js:18 WebSocket connection to 'ws://localhost:3000/socket/?EIO=3&transport=websocket&sid=z12aPFY1Uxzl1qh8AAAB' failed: Error during WebSocket handshake: Unexpected response code: 400`

**Comment by rubenhak** [comment]
@nadworny, glad it worked. We are working on areas to shorted initial load time.

WebSocket error is unrelated, but will try to eliminate it to avoid confusion. 

Keeping this open for tracking purposes.

**Comment by guillem-riera** [comment]
Just a short update:
I am using istio (demo settings, with auto injection in the kubevious namespace) in my minikube cluster.

After removing the release with helm, and removing the auto injection in the namespace, I was able to run kubevious without errors (the latest helm release that can be found using `helm search repo kubevious -l`, which was `0.7.22`).

Might it be strictly related to istio?

**Comment by rubenhak** [comment]
@guillem-riera, thanks for the update. We are preparing a major release. Would be able to take a look to this after that.

Meanwhile, it would be helpful to have Istio configs you used to debug this further.

**Comment by kubevious** [comment]
Folks, can you try the latest version and let us know if you are still experiencing this?https://github.com/kubevious/kubevious/releases/tag/v0.8

@guillem-riera, yes, if service mesh is configured in such a way that would prevent ui to backend communication that could be the reason.

**Comment by anerathil** [comment]
Hey, 
i just deployed the Helm chart (version 0.8.15) in a namespace with Istio + sidecar injection (no connection restrictions configured) and waited a few minutes.

I also get the 426 (Upgrade Required). There are no errors logs in the pods.
Tried executing a curl (curl -v $BACKEND_URL) in the Kubevious ui pod - http 200 was returned.

My Istio config is:

```
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  values:
    global:
      proxy:
        autoInject: enabled
      useMCP: false
      tracer:
        zipkin:
          address: zipkin.istio-system.svc.cluster.local:9411
  addonComponents:
    prometheus:
      enabled: true
    kiali:
      enabled: false
    grafana:
      enabled: false
    tracing:
      enabled: true

  components:
    ingressGateways:
      - name: istio-ingressgateway
        enabled: true
      - name: cluster-local-gateway
        enabled: true
        label:
          istio: cluster-local-gateway
          app: cluster-local-gateway
        k8s:
          service:
            type: ClusterIP
            ports:
            - port: 15020
              name: status-port
            - port: 80
              name: http2
            - port: 443
              name: https
```

Any ideas?

**Comment by rubenhak** [comment]
ahh... Istio... :)
TBH, we haven't got our hands on Istio yet, so its hard to answer right now. But I'm happy to help. Would you like to join the slack channel to do remote debugging together? I think we would be able to find the solution. Two sets of eyes is better than one. Here it is? [https://kubevious.io/slack](https://kubevious.io/slack)

**Comment by kubevious** [comment]
@nadworny, @guillem-riera, this was fixed in version **0.9.2**.
```
$ helm upgrade --atomic -i kubevious kubevious/kubevious --version 0.9.2 -n kubevious 
```

@anerathil, thanks for helping to root cause and validate the fix.
